Kommentare Marc

bh0fer opened the pull request.

mcdelarosa committed:
Delete Welcome

mcdelarosa committed:
Create README.md

mcdelarosa committed:
Update README.md

mcdelarosa committed:
Delete README.md

mcdelarosa committed:
Create README.md

mcdelarosa committed:
Code Beispiel hinzugefügt

mcdelarosa committed:
Blockbeitrag

bh0fer commented on blog/2022-08-26-blog.md:
```diff
@@ -0,0 +1,23 @@
+#   **Wiedereinstieg in Python**
+
+Das heutige Thema war Python/Tigerjython was mir nicht wirklich gefällt, da das nur eine Bibliothek,
+die keiner im Alltag braucht und nutzlos ist.
```
Das Thema war Turtlegrafik, welches eine Bibliothek ist... 
(Tigerjython ist eine in Java geschriebene Implementation von Python, welche zusätzliche Sprachkonstrukte wie `repeat` erlaubt. (Wird ausserhalb des Unterrichts aber nicht eingesetzt...))

mcdelarosa committed:
Code Beispiel hinzugefügt

mcdelarosa committed:
Vieles gemacht und mit dem Spiel angefangen

mcdelarosa committed:
Die Hausaufgaben habe ich erledigt

mcdelarosa committed:
Die Aufgabe mit dem Rechteck habe ich versucht möglichst viel zu optimieren und schön zu programmieren, was mir dann (durch ein wenig Hilfe) gelang

mcdelarosa committed:
Hangman wurde noch weiterentwickelt doch wurde noch nicht abgeschlossen.

mcdelarosa committed:
Hangman wurde noch weiterentwickelt doch wurde noch nicht abgeschlossen.

mcdelarosa committed:
Hangman wurde noch weiterentwickelt doch wurde noch nicht abgeschlossen.
Habe Zeilen gefunden, welche die Gründe sein könnten für die ganzen Problemen, jedoch noch nicht behoben...

bh0fer commented on exercises/fkt/sternchen.py:
```diff
@@ -0,0 +1,22 @@
+width = input("Enter a width for the rectangle")
+y = int(height)
+x = int(width)
+def constructionOfBase(number):
+    for repetitions in range(number, 0, -1):
+        print("*", end="")
+    print("")
```
Exemplarisch bei dieser Funktion, gilt aber überall:
1. Das Programm Funktioniert, top! 
2. Namensgebung ist eine Philosophie-Frage, die Python-Community verwendet `_` als Trennzeichen zwischen Wörtern aber bspw. JavaScript macht es mit Gross-Kleinbuchstaben.
3. Was ist der Vorteil in Ihrem Code von `range(number, 0, -1)` gegenüber `range(number)`? --> Wenn möglich die einfachste und am besten verständliche Version verwenden...

```suggestion
def construction_of_base(number):
    for repetitions in range(number):
        print("*", end="")
    print("")
```

mcdelarosa replied:
Anfangs dachte ich, dass man alles definieren mussten bei `range()`, deshalb hatte ich auch dies so geschrieben...
Und wegen der Bennenung, das werde ich für künftig beachten, ich wusste von der Vielfalt an Namensgebung für verscheidenen Programmiersprachen nicht.
Danke!

mcdelarosa committed:
Ich habe das Hangman endlich abgeschlossen!!!

mcdelarosa committed:
Die Hausaufgauben habe ich erledigt: Hangman ist endlich spielbar und komplett abgeschlossen. Und die Kästchen für Numtrip wurden auch programmiert.

mcdelarosa committed:
Die Hausaufgauben habe ich erledigt: Hangman ist endlich spielbar und komplett abgeschlossen. Und die Kästchen für Numtrip wurden auch programmiert.

mcdelarosa committed:
Hangman in python

mcdelarosa committed:
Numtrip

The game is still in development...

mcdelarosa committed:
Links for the Website

mcdelarosa committed:
Website

Website with docusaurus

mcdelarosa committed:
Website

Website with docusaurus.
Working on a better website.

mcdelarosa committed:
Website

Website with docusaurus.
Colors where personalized.

mcdelarosa committed:
Website

Website with docusaurus.
Colors where personalized.

mcdelarosa committed:
Website

Website with docusaurus.
Colors where personalized.

mcdelarosa committed:
Website

Website with docusaurus.
Colors where personalized.

mcdelarosa committed:
Website

Website with docusaurus.
Colors where personalized.
The icon was changed

mcdelarosa committed:
Website

Website with docusaurus.
Colors where personalized.
The icon was changed

mcdelarosa committed:
Website

Website with docusaurus.
Colors where personalized.
The icon was changed

mcdelarosa committed:
Website

Website with docusaurus.
Colors where personalized.
The icon was changed

mcdelarosa committed:
Rectangle-generator

The function constructionOfBase was improved.

mcdelarosa committed:
Website

My website is now personalized

bh0fer commented on blog/2022-08-26-blog.md:
```diff
@@ -0,0 +1,23 @@
+#   **Wiedereinstieg in Python**
```
Bei einem Titel (mit `#`, `##` oder `###`...) ist das fett markieren mit `**`. nicht nötig

```suggestion
#   Wiedereinstieg in Python
```

mcdelarosa committed:
Website

My website is now personalized

mcdelarosa committed:
Website

My website is now personalized

mcdelarosa committed:
Website

My website is now personalized

mcdelarosa committed:
Website

My website is now personalized

mcdelarosa committed:
Website

My website is now personalized

mcdelarosa committed:
Website

My website is now personalized

mcdelarosa committed:
Website

My website is now personalized

mcdelarosa committed:
Website

My website is now personalized

mcdelarosa committed:
Blog

Neues geschrieben!

mcdelarosa committed:
Numtrip

-New Design
-Gameloop
-Code has been simplified & reduced

bh0fer commented on numtrip/game.py:
```diff
@@ -31,11 +33,26 @@ def rows (self):
         Numtrip.coordinateY = Numtrip.coordinateY + 1
+    def analyse(self):
+        coordinate = Numtrip.wanted.strip()
+        """
```
Cool, Sie verwenden bereits Klassen. Kennen Sie den Unterschied zwischen Instanzvariablen und Klassenvariablen?

Wozu braucht man das `self`? Warum plötzlich `Numtrip.wanted`?

--> Wenn Sie den Klassen-basierten Ansatz wählen, so müssen Sie die verwendeten Konzepte beim Fachgespräch erklären können.

mcdelarosa replied:
Klassenvariablen können alles enthalten (soviel ich weiss) und zwar: Funktionen, Variablen, Listen(Arrays) und Loops.
Die Klassenvariabeln dienen als eine Art Liste, und/oder Vorlagen für die Instanzvariabeln. Wenn man nichts in den Instanzvariabeln definiert, so wird sie eine Kopie von der Klassenvariable, wird aber etwas hingeschrieben so ändern sich die gewollten Werte, doch die Klassenvariable bleibt unverändert.

Das habe ich eigentlich neugeschrieben, jedoch will GitHub die neuste Version nicht hochladen (es ist nicht das erste Mal und auch nicht das einzige Dokument)...
`self `nutzt man hier für das Gleiche wie `Numtrip`, das greift auf die Werte , welche in der Klassenvariable `Numtrip `definiert wurden in diesem Fall `wanted`, aber `Numtrip.wanted` (`self.wanted`) ist eine Instanzvariabel.

mcdelarosa committed:
Blog

Probleme und weitere Probleme...

mcdelarosa committed:
Numtrip

New:
-Changes the numbers
To do:
-Different colors for different numbers

mcdelarosa committed:
Numtrip

New:
-Changes the numbers
To do:
-Different colors for different numbers

mcdelarosa committed:
Delete Numtrip.py

mcdelarosa committed:
Numtrip

New:
-Changes the numbers
To do:
-Different colors for different numbers
-Fill the gaps
-To duplicate the choosen value

mcdelarosa committed:
Numtrip

New:
-Changes the numbers
To do:
-Different colors for different numbers
-Fill the gaps
-To duplicate the chosen value

mcdelarosa committed:
Merge remote-tracking branch 'origin/main'

mcdelarosa committed:
Numtrip

New:
-Changes the numbers
To do:
-Different colors for different numbers
-Fill the gaps
-To duplicate the chosen value

mcdelarosa committed:
Numtrip

New:
-Changes the numbers and their places (via gravitation)
To do:
-Different colors for different numbers
-Fill the gaps with numbers of the base 2

mcdelarosa committed:
Numtrip

New:
-Changes the numbers and their places (via gravitation)
-Different colors for different numbers
-Fill the gaps with numbers of the base 2

mcdelarosa committed:
Website

My private Blog

mcdelarosa committed:
Settings

mcdelarosa committed:
Numtrip

New:
-Random board

mcdelarosa committed:
Blog

mcdelarosa committed:
Excersises

Playing with data in python with json

mcdelarosa committed:
gitignore

mcdelarosa committed:
Blog

Mein Blog

mcdelarosa committed:
Update README.md

mcdelarosa committed:
Update README.md

mcdelarosa committed:
Update README.md

mcdelarosa committed:
Update README.md

mcdelarosa committed:
Update README.md

mcdelarosa committed:
Update README.md

mcdelarosa committed:
Update README.md

mcdelarosa committed:
Update README.md

mcdelarosa committed:
Update README.md

mcdelarosa committed:
Update README.md

mcdelarosa committed:
Numtrip

New:
-Score
-Duplication only if there are the same numbers
-Game-end
-Bugs fixed
-Code is simplificated

mcdelarosa committed:
Merge remote-tracking branch 'origin/main'

mcdelarosa committed:
Blog

Rename

mcdelarosa committed:
Blog

Rename

mcdelarosa committed:
Blog

Rename

mcdelarosa committed:
Blog

Rename

mcdelarosa committed:
Blog

Rename

mcdelarosa committed:
Blog

Letztes Blog schrieben und letzte Bearbeitungen an jeden Blogeintrag.

mcdelarosa committed:
Blog

Letztes Blog schrieben und letzte Bearbeitungen an jeden Blogeintrag.

mcdelarosa committed:
Readme

Updated some little things, just to make everything easier for everyone and to look better :)

mcdelarosa committed:
Blog

Kleine Änderung für den letzten Eintrag

mcdelarosa committed:
Blog

Namensänderungen

mcdelarosa committed:
Blog

Namensänderungen und Informationen zu den Bildern

mcdelarosa committed:
Blog

Namensänderungen und Informationen zu den Bildern

mcdelarosa committed:
Blog

Namensänderungen und Informationen zu den Bildern

mcdelarosa committed:
Blog

Namensänderungen und Informationen zu den Bildern

mcdelarosa committed:
Blog

Namensänderungen und Absätze

mcdelarosa committed:
Blog

Deskription hinzugefügt

mcdelarosa committed:
Blog

Top-Down-Entwurf

mcdelarosa committed:
Blog

Abgabe Blog

mcdelarosa committed:
fix path
